src/rdrand.h

```diff
@@ -29,6 +29,36 @@
 #ifndef _RDRAND_H_
 #define _RDRAND_H_
 
+#if WITH_SSL
+#include <openssl/crypto.h>
+#if OPENSSL_VERSION_NUMBER >= 0x10101000L
+#define HAVE_OSSL_SECURE_RAND_BYTES 1
+#define HAVE_SECURE_RAND_BYTES      1
+#include <openssl/rand.h>
+#endif
+#endif
+#ifndef HAVE_OSSL_SECURE_RAND_BYTES
+#if HAVE_GETENTROPY || defined(_WIN32)
```

Comment:
We should modify this condition correctly based on the function present instead of relying 1 part of the condition of the feature and another part on the OS. All windows might not have Secure rand bytes as rand_s is not uniformly available in all windows.